src/p2p/Join/index.ts

```diff
@@ -725,11 +742,54 @@ export function sendRequests(): void {
         ]),
         true
       )
-      nestedCountersInstance.countEvent('p2p', `saved join request and gossiped to network`)
-      /* prettier-ignore */ if (logFlags.verbose) console.log(`saved join request and gossiped to network`)
+      nestedCountersInstance.countEvent('p2p', `join:sendRequests: saved join request and gossiped to network`)
+      /* prettier-ignore */ if (logFlags.p2pNonFatal) console.log(`join:sendRequests: saved join request and gossiped to network`)
     }
     queuedJoinRequestsForGossip = []
   }
+
+  console.log('Join:sendRequests: queuedUnjoinReuqests', queuedUnjoinReuqests)
+  if (queuedUnjoinReuqests.length > 0) {
+    for (const unjoinRequest of queuedUnjoinReuqests) {
+      console.log('Join:sendRequests: unjoinRequest', unjoinRequest)
+
+      // TO-DO: address this following:
+      // processing it again to make sure it is still valid but
+      // we already processed it in the unjoin route; can we just gossip it?
+      //
+      // there an issue i've thought of: processNewUnjoinRequest automatically 
+      // adds the unjoin request to the node's unjoin map. Since we call it last cycle in the unjoin handler but dont gossip it,
```

Comment:
I think i already handled this by removing the unjoin request the cycle that we receive it. line ref is 229 inside src/p2p/Join/routes.ts